### PR TITLE
Refactoring of Copy-DbaCredential and Export-DbaCredential

### PR DESCRIPTION
**Pull request by andreasjordan**
General refactoring to be able to open DAC inside of Copy-DbaCredential.
Fixed some issues with QUOTENAME of credential name

Still open: Opening DAC in Export like in Copy. Will have to save all the $server variables to be able to disconnect at the end.

### CODE REVIEW

File: functions/Copy-DbaCredential.ps1 [hunk 1]
```diff
@@ -117,37 +118,54 @@ function Copy-DbaCredential {
...
-                $destServer.Credentials.Refresh()
-                $credentialName = $credential.Name
+            $destServer.Credentials.Refresh()
+            foreach ($cred in $credentialList) {
+                $credentialName = $cred.Name
 
                 $copyCredentialStatus = [pscustomobject]@{
-                    SourceServer      = $sourceServer.Name
-                    DestinationServer = $destServer.Name
+                    SourceServer      = $sourceServer.DomainInstanceName
```
Comment (potatoqualitee) [inline functions/Copy-DbaCredential.ps1:167]: Should this be ComputerName instead? I feel like there are sometimes problems with DomainInstanceName. Perhaps it's not always filled out or accurate.

Comment (wsmelton) [reply functions/Copy-DbaCredential.ps1:167]: `SourceServer` is used for the output object in all the Copy commands @potatoqualitee

Comment (wsmelton) [reply functions/Copy-DbaCredential.ps1:167]: Oh, you are referring to property name? If that, yes we use ComputerName for the value in other commands.

Comment (potatoqualitee) [reply functions/Copy-DbaCredential.ps1:167]: Thank you for double checking 😊

Comment (wsmelton) [reply functions/Copy-DbaCredential.ps1:167]: Also looks like `Connect` object does checks for populating `ComputerName`

https://github.com/dataplat/dbatools/blob/3d8d330f001853268c78aa0eebb2aef10ae9630c/functions/Connect-DbaInstance.ps1#L1573-L1592

Comment (andreasjordan) [reply functions/Copy-DbaCredential.ps1:167]: No, we don't need the computername here, but "SqlInstance". This is the output from the currently released code:

image

And we mostly fill "SqlInstance" using ".DomainInstanceName", just search for "SqlInstance  = $server.DomainInstanceName" in the code.

We can not use ".Name" of "$server", as this has the prefix "ADMIN:" when using a DAC - and I think we don't want to have that in the output.

Comment (potatoqualitee) [reply functions/Copy-DbaCredential.ps1:167]: ahhhh make sense, thanks

**Comment by potatoqualitee** [comment]
Thank you 🙇🏼

**Pull request merged**